src/plugins/workspace/public/utils.ts

```diff
@@ -223,6 +226,20 @@ export const getDataSourcesList = (client: SavedObjectsStart['client'], workspac
     });
 };
 
+// If all connected data sources are serverless, will only allow to select essential use case.
+export const getIsOnlyAllowEssentialUseCase = async (client: SavedObjectsStart['client']) => {
+  const allDataSources = await getDataSourcesList(client, ['*']);
+  if (allDataSources.length > 0) {
+    const serverlessDataSource = allDataSources.filter(
+      (ds) => ds?.auth?.credentials?.service === SigV4ServiceName.OpenSearchServerless
+    );
+    if (serverlessDataSource.length === allDataSources.length) {
+      return true;
+    }
+  }
+  return false;
```

Comment:
So for non dashboard admin, if all permitted data sources are serverless, then the use case can only be changed to   essential use case when workspace update, is that a expected behavior? There may includes some unassigned data sources that not serverless.